### DISCUSSION TITLE
Benchmarking with num_search_worker

### DISCUSSION THREAD

**Discussion by godfeed27** [google/or-tools#4471] [CP-SAT questions]
Version: v9.10.4067
Language: Python

Solver: CP-SAT

I did a preliminary evaluation of my problem by running it many times with different options for num_search_worker and got the results as shown in the picture. Why is it that when using the most threads, most of the results are worse than using fewer threads (on same CPU)? How does num_search_worker work?![Image](https://github.com/user-attachments/assets/b53684be-984b-4dbe-9f7f-4f2a84a9863f)

**Comment by lperron** [comment]
your cpus have 8 and 6 real cores. 48 workers make 6 or 8 workers run on each code, slowing down everything.
